importer.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type Importer struct {
+	e *tfexec.Terraform
+	c *tfe.Client
+}
+
+func NewImporter(tf *tfexec.Terraform, token string, host string) (*Importer, error) {
+	tfc, err := tfe.NewClient(&tfe.Config{
+		Token:   token,
+		Address: fmt.Sprintf("https://%s", host),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &Importer{
+		tf,
+		tfc,
+	}, nil
+}
+
+func (tf *Importer) ImportWorkspace(ctx context.Context, name string, organization string, opts ...tfexec.ImportOption) error {
+	address := fmt.Sprintf("tfe_workspace.workspace[\"%s\"]", name)
+
+	state, err := tf.e.Show(context.Background())
+	if err != nil {
+		log.Fatalf("Failed to read state: %s", err)
+	}
+
+	found := false
+	if state.Values == nil {
+		fmt.Println("No state found, importing")
+	} else {
+		for _, r := range state.Values.RootModule.Resources {
```

Comment:
Avoid complex for loop controls where possible. Similar to above, a `HasResource(state *tfjson.State, addr string) bool` method will both help separate UI from lower level logic and make the code easier to read. Rather than `found = true; break` you can just `return true` and `return false` outside the loop.